packages/Box/Box.jsx

```diff
@@ -1,70 +1,113 @@
 import React from 'react'
+import styled from 'styled-components'
 import PropTypes from 'prop-types'
 
-import joinClassNames from '../../shared/utils/joinClassNames'
-import capitalize from '../../shared/utils/capitalize'
-import { deprecate } from '../../shared/utils/warn'
+import { media } from '@tds/core-responsive'
 
-import styles from './Box.modules.scss'
+const spacing = {
+  mobile: {
+    1: '0.25rem',
+    2: '0.5rem',
+    3: '1rem',
+    4: '1.5rem',
+    5: '2rem',
+    6: '2.5rem',
+    7: '3rem',
+    8: '4rem',
+  },
+  desktop: {
+    1: '0.25rem',
+    2: '0.5rem',
+    3: '1rem',
+    4: '2rem',
+    5: '3rem',
+    6: '4rem',
+    7: '4.5rem',
+    8: '6rem',
+  },
+}
 
-const getClassName = (spacing, location, scale) => {
-  if (!scale) {
-    return undefined
-  }
-  return styles[`${location}${capitalize(spacing)}-${scale}`]
+const boxSpacing = (level, f, size) => {
+  return f(spacing[size][level])
 }
 
-const getBetweenClasses = (scale, inline) => {
-  if (!scale) {
-    return undefined
+const betweenStyles = (between, inline, size) => {
+  if (between === 'space-between') {
+    return { justifyContent: 'space-between' }
   }
 
-  const direction = inline ? 'Right' : 'Bottom'
-  return joinClassNames(
-    scale === 'space-between' ? styles.spaceBetween : styles[`between${direction}Margin-${scale}`],
-    inline ? styles.inline : styles.stack
+  return boxSpacing(
+    between,
+    space => ({
+      '> *:not(:last-child)': {
+        ...(inline ? { marginRight: space } : { marginBottom: space }),
+      },
+    }),
+    size
+  )
+}
+
+const horizontalStyles = (horizontal, size) => {
+  return boxSpacing(
+    horizontal,
+    space => ({
+      paddingLeft: space,
+      paddingRight: space,
+    }),
+    size
+  )
+}
+
+const verticalStyles = (vertical, size) => {
+  return boxSpacing(
+    vertical,
+    space => ({
+      paddingTop: space,
+      paddingBottom: space,
+    }),
+    size
+  )
+}
+
+const insetStyles = (inset, size) => {
+  return { ...verticalStyles(inset, size), ...horizontalStyles(inset, size) }
+}
+
+const belowStyles = (below, size) => {
+  return boxSpacing(
+    below,
+    space => ({
+      marginBottom: space,
+    }),
+    size
   )
 }
 
+const StyledBox = styled.div.attrs(({ className, tag }) => {
+  return { className, as: tag }
+})(({ inline, between, horizontal, vertical, inset, below }) => ({
+  display: between ? 'flex' : 'block',
```

Comment:
Instead of spreading, we should pass each function as an argument to styled(). This way, 'styled-components' will deeply merge all properties.